Client/Client.php

```diff
@@ -140,77 +119,52 @@ protected function fillDataFromXML(SettlementResponse $settlementResponse, $xml)
-            'POST',
-            $data
-        );
-        $response = $this->request($request);
+        $this->authenticationStrategy->authenticate($parameter, AuthenticationStrategy::SETTLEMENT);
 
-        $this->logger->info("Response:\n" . $response->getContent());
+        $response = $this->sendRequest($parameter);
 
         if ($response->getStatus() != 200) {
-            $this->logger->critical('Datatrans: request failed with statuscode: {statuscode}!', array('statuscode' => $response->getStatusCode()));
-            throw new \Exception('Datatrans: request failed with statuscode: ' . $response->getStatus() . '!');
+            $this->logger->critical('Datatrans: request failed with statuscode: {statuscode}!', ['statuscode' => $response->getStatusCode()]);
+            throw new Exception('Datatrans: request failed with statuscode: ' . $response->getStatus());
         }
+
         return $response->getContent();
     }
 
-    public function request(Request $request)
+    public function sendRequest(ParameterInterface $parameter)
     {
         if (!extension_loaded('curl')) {
-            throw new \RuntimeException('The cURL extension must be loaded.');
+            throw new RuntimeException('The cURL extension must be loaded.');
         }
 
-        $curl = curl_init();
-        curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
-        curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
-        curl_setopt($curl, CURLOPT_URL, $request->getUri());
-        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($curl, CURLOPT_HEADER, true);
-
-        // add headers
-        $headers = array();
-        foreach ($request->headers->all() as $name => $value) {
-            if (is_array($value)) {
-                foreach ($value as $subValue) {
-                    $headers[] = sprintf('%s: %s', $name, $subValue);
-                }
-            } else {
-                $headers[] = sprintf('%s: %s', $name, $value);
-            }
-        }
-        if (count($headers) > 0) {
-            curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
-        }
+        $data = array('xmlRequest' => $this->fillXMLFromData($parameter->getData()));
 
-        // set method
-        $method = strtoupper($request->getMethod());
-        if ('POST' === $method) {
-            curl_setopt($curl, CURLOPT_POST, true);
-            if (!$request->headers->has('Content-Type') || 'multipart/form-data' !== $request->headers->get('Content-Type')) {
-                $postFields = http_build_query($request->request->all());
-            } else {
-                $postFields = $request->request->all();
-            }
+        $this->logger->info($parameter->getRequestUrl());
+        $this->logger->info("Request:\n" . var_export($data, true));
+
+        $curlOpts = [
+            CURLOPT_URL => $parameter->getRequestUrl(),
+            CURLOPT_POST => true,
+            CURLOPT_SSL_VERIFYHOST => false,
```

Comment:
should be set to `2` according to https://www.php.net/manual/en/function.curl-setopt.php